lib/Runtime/Language/JavascriptConversion.inl

```diff
@@ -203,4 +203,40 @@ namespace Js {
        return SameValueCommon<true>(aValue, bValue);
    }
 
+   template <bool allowNegOne>
+   inline Var JavascriptConversion::TryCanonicalizeAsTaggedInt(Var value)
+   {
+       if (TaggedInt::Is(value))
+       {
+           return (allowNegOne || value != TaggedInt::ToVarUnchecked(-1))
+               ? value
+               : nullptr;
+       }
+
+       if (!JavascriptNumber::Is(value))
+       {
+           return nullptr;
+       }
+
+       double doubleVal = JavascriptNumber::GetValue(value);
+       int32 intVal = 0;
+
+       if (!JavascriptNumber::TryGetInt32Value<true>(doubleVal, &intVal))
+       {
+           return nullptr;
```

Comment:
>nullptr [](start = 18, length = 7)

returning nullptr or false in this function seems to be inconsistent with return type.

Reply:
oops good point

---
In reply to: [174315122](https://github.com/Microsoft/ChakraCore/pull/4816#discussion_r174315122) [](ancestors = 174315122)

Reply:
i meant for nullptr in the failure case (similar to what we do for the TryFromVar methods)

---
In reply to: [174315376](https://github.com/Microsoft/ChakraCore/pull/4816#discussion_r174315376) [](ancestors = 174315376,174315122)